### PR TITLE
Add a section about {dotnet_root} requirements

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
## Summary

Clarifying requirements for {dotnet_root} folder, to ensure correct SxS implementation of .NET packaging.

### CODE REVIEW

File: docs/core/distribution-packaging.md [hunk 1]
```diff
@@ -99,6 +99,8 @@ The **shared** folder contains frameworks. A shared framework provides a set of
 
 The folders marked with `(*)` are used by multiple packages. Some package formats (for example, `rpm`) require special handling of such folders. The package maintainer must take care of this.
 
+- (18) **{dotnet_root}** is a shared root for all .NET major and minor versions. As a consequence, you will see all diffrent runtime versions, for all .NET major and minor versions, under a single folder on user's machine - `{dotnet_root}/shared/Microsoft.NETCore.App`. The same is true for all other folders that contain version sub-folders, i.e. `{dotnet_root}/shared/sdk`. It is strongly advised that the name of `{dotnet_root}` folder be version agnostic, i.e. simply `dotnet`.
```
Comment (MichaelSimons) [inline docs/core/distribution-packaging.md:102]: I'm not finding the `(18)` reference in the above layout.  The numbers here are cross references to the above layout.  I think 18 needs to be added.

Comment (MichaelSimons) [inline docs/core/distribution-packaging.md:102]:
```suggestion
- (18) **{dotnet_root}** is a shared root for all .NET major and minor versions. For example, if multiple runtimes are installed, they would share the **{dotnet_root}** folder, `{dotnet_root}/shared/Microsoft.NETCore.App/6.0.11` and `{dotnet_root}/shared/Microsoft.NETCore.App/7.0.0`. It is advised that the name of `{dotnet_root}` folder be version agnostic, i.e. simply `dotnet`.
```

Comment (NikolaMilosavljevic) [reply docs/core/distribution-packaging.md:102]: I see, let me check how to update the table. I was thinking of adding this new section to be the first one, but feared the big diff due to renumbering. I'll see if {0} is also possible.

Comment (NikolaMilosavljevic) [reply docs/core/distribution-packaging.md]: I've incorporated this in the second commit. I've renamed {18} to {0} and updated the layout accordingly.

File: docs/core/distribution-packaging.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ As .NET 5 (and .NET Core) and later versions become available on more and more p
 When installed, .NET consists of several components that are laid out as follows in the file system:
 
 ```
-{dotnet_root}                                     (*)
+{dotnet_root}                    (0)
```
Comment (MichaelSimons) [inline docs/core/distribution-packaging.md:19]: I think the asterias should be maintained as the `dotnet_root` is used by multiple packages.

```suggestion
{dotnet_root}                    (0)               (*)
```

**Review by MichaelSimons — APPROVED** [review]

File: docs/core/distribution-packaging.md [hunk 3]
```diff
@@ -63,6 +63,8 @@ When installed, .NET consists of several components that are laid out as follows
         └── dotnet               (10)
 ```
 
+- (0) **{dotnet_root}** is a shared root for all .NET major and minor versions. For example, if multiple runtimes are installed, they would share the **{dotnet_root}** folder, `{dotnet_root}/shared/Microsoft.NETCore.App/6.0.11` and `{dotnet_root}/shared/Microsoft.NETCore.App/7.0.0`. It is advised that the name of `{dotnet_root}` folder be version agnostic, i.e. simply `dotnet`.
```
Comment (gewarren) [inline docs/core/distribution-packaging.md:66]:
```suggestion
- (0) **{dotnet_root}** is a shared root for all .NET major and minor versions. If multiple runtimes are installed, they share the **{dotnet_root}** folder, for example, `{dotnet_root}/shared/Microsoft.NETCore.App/6.0.11` and `{dotnet_root}/shared/Microsoft.NETCore.App/7.0.0`. The name of the `{dotnet_root}` folder should be version agnostic, that is, simply `dotnet`.
```

**Pull request merged**